src/cpp/src/whisper/pipeline_static.cpp

```diff
@@ -189,14 +196,14 @@ void zero_past_key_values(ov::InferRequest& request) {
 }
 
 void prepare_decoder_with_past(ov::InferRequest& decoder_with_past, ov::InferRequest& decoder, const size_t init_ids_size) {
-    // NB: Prepare attetion mask to be in a format [0, 0, 0, 1, 1, 1, 1, ..., 0, 1]
-    // Mask should be inverted for decoder_with_past 
-    auto attention_mask = decoder_with_past.get_tensor("attention_mask");
-    auto* attention_mask_ptr = attention_mask.data<ov::float16>();
-    std::fill(attention_mask_ptr, attention_mask_ptr + init_ids_size, 0);
-    std::fill(attention_mask_ptr + init_ids_size, attention_mask_ptr + attention_mask.get_size() - 2, 1);
-    attention_mask_ptr[attention_mask.get_size() - 2] = 0;
-    attention_mask_ptr[attention_mask.get_size() - 1] = 1;
+    // NB: Prepare attetion mask to be in a format [0, 0, 0, 1, 1, 1, ..., 1, 0, 1]
+    auto padded_attention_mask = decoder_with_past.get_tensor("attention_mask");
+    auto* padded_mask_data = padded_attention_mask.data<float>();
+    std::fill(padded_mask_data, padded_mask_data + init_ids_size, 0);
```

Comment:
Shouldn't we also add some assert checks on sizes?
```C++
OPENVINO_ASSERT(padded_attention_mask.get_size() >= init_ids_size);
OPENVINO_ASSERT(padded_attention_mask.get_size() >= 2);
```

Reply:
Added